tests/unit/customizations/test_commands.py

```diff
@@ -136,6 +142,25 @@ def test_expected_events_are_emitted_in_order(self):
             'before-building-argument-table-parser.s3.foo'
         ])
 
+    def test_expected_events_are_emitted_in_order_with_event_name(self):
+        driver = CLIDriver(session=self.session)
+        self.emitter.register(
+            'building-command-table.s3', self.inject_command_with_event_name)
+
+        driver.main('s3 foo'.split())
+
+        self.assert_events_fired_in_order([
+            'building-command-table.main',
+            'building-top-level-params',
+            'top-level-args-parsed',
+            'session-initialized',
+            'building-command-table.s3',
+            # here should be use event_name
+            'building-command-table.foo-command',
+            'building-arg-table.foo',
```

Comment:
Whatever we decide for the `building-command-table` event we should probably apply to it to this event as well because it may exhibit similar issues with collisions in the future.